src/obstacle_avoidance/src/go_through_hole.cpp

```diff
@@ -0,0 +1,24 @@
+/*
+ * Created By: Gareth Ellis
+ * Created On: July 16th, 2016
+ * Description: An example node that subscribes to a topic publishing strings,
+ *              and re-publishes everything it receives to another topic with
+ *              a "!" at the end
+ */
+
+#include <GoThroughHole.h>
+
+
+int main(int argc, char **argv){
+    // Setup your ROS node
+    std::string node_name = "my_node";
```

Comment:
you should change the string name to be less generic. Maybe "go_through_hole_node".